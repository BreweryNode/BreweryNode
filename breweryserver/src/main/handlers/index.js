require('./coldwaterhandler');
require('./hotwaterhandler');
require('./boilerhandler');
require('./manifoldhandler');
require('./heatexchangerhandler');
require('./mashtunhandler');
