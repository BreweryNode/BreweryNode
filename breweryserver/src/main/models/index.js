const Sequelize = require('sequelize');
const env = process.env.NODE_ENV || 'development';
const config = require('../config/config.js')[env];
let db = {};
let sequelize;

if (config.use_env_variable) {
  sequelize = new Sequelize(process.env[config.use_env_variable]);
} else {
  sequelize = new Sequelize(config.database, config.username, config.password, config);
}

let model = sequelize.import('boiler', require('./boiler'));
db.Boiler = model.single;

model = sequelize.import('coldwater', require('./coldwater'));
db.ColdWater = model.single;

model = sequelize.import('hotwater', require('./hotwater'));
db.HotWater = model.single;

db.MashTun = sequelize.import('mashtun', require('./mashtun'));

db.Manifold = sequelize.import('manifold', require('./manifold'));

model = sequelize.import('heatexchanger', require('./heatexchanger'));
db.HeatExchanger = model.single;

Object.keys(db).forEach(function(modelName) {
  if (db[modelName].associate) {
    db[modelName].associate(db);
  }
});

db.sequelize = sequelize;
db.Sequelize = Sequelize;

module.exports = db;
