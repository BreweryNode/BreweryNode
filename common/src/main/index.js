exports.mq = require('./mq');
exports.logutil = require('./logutil');
exports.sensorutil = require('./sensorutil');
exports.mutatorutil = require('./mutatorutil');
exports.messageutil = require('./messageutil');
