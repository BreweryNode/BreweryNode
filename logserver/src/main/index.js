var Promise = require('bluebird');
var models;
let mq = require('brewerynode-common').mq;
var logutil = require('brewerynode-common').logutil;
var winston = require('winston');

function startDB() {
  return new Promise(function(resolve, reject) {
    models = require('./models');
    console.log('Syncing database');
    models.sequelize
      .sync({ force: false })
      .then(() => {
        console.log("Database sync'd");
        resolve();
      })
      .catch(err => {
        console.warn(err);
        reject(err);
      });
  });
}

function handleNewLog(msg) {
  return new Promise(function(resolve, reject) {
    let lLog = JSON.parse(msg.content.toString());
    if (!lLog.hasOwnProperty('message') || !lLog.hasOwnProperty('level')) {
      console.warn('Bad DTO: ' + JSON.stringify(lLog));
      reject();
      return;
    }
    models.Log.create(lLog)
      .then(() => {
        winston.log(lLog.level, lLog.message);
        resolve();
      })
      .catch(err => {
        console.error('Error saving log:\n' + err);
        reject();
      });
  });
}

function startMQ() {
  return new Promise(function(resolve, reject) {
    console.log('Connecting to MQ');
    mq
      .connect(process.env.MQ_ADDRESS, 'amq.topic')
      .then(() => {
        console.log('MQ Connected');
        return Promise.all([mq.recv('logserver', 'logging.v1.#', false, handleNewLog)]);
      })
      .then(() => {
        console.log('MQ Listening');
        resolve();
      })
      .catch(err => {
        console.warn(err);
        reject(err);
      });
  });
}

async function main() {
  console.log('Starting');
  await Promise.all([startDB(), startMQ()]);
  logutil.info('Log server started');
}

main();
