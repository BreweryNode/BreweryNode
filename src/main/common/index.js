exports.mq = require('./mq');
exports.logutil = require('./logutil');
