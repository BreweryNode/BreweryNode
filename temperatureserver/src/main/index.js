const Promise = require('bluebird');
var models;
let mq = require('brewerynode-common').mq;
var logutil = require('brewerynode-common').logutil;

function startDB() {
  return new Promise(function(resolve, reject) {
    models = require('./models');
    logutil.silly('Syncing database');
    models.sequelize
      .sync({ force: false })
      .then(() => {
        logutil.silly("Database sync'd");
        resolve();
      })
      .catch(err => {
        logutil.warn(err);
        reject(err);
      });
  });
}

function addTempertureProbe(pProbe) {
  return new Promise(function(resolve, reject) {
    models.Temperature.create(pProbe)
      .then(() => {
        logutil.info(
          'Created temperature probe: ' + pProbe.name + ' with mac: ' + pProbe.mac
        );
        resolve();
      })
      .catch(err => {
        logutil.error('Error creating temperature probe:\n' + err);
        reject();
      });
  });
}

function handleCreateNew(msg) {
  return new Promise(function(resolve, reject) {
    let lDTO;
    try {
      lDTO = JSON.parse(msg.content.toString());
    } catch (err) {
      logutil.error('Error parsing message:\n' + err);
      reject();
      return;
    }
    if (
      !Object.prototype.hasOwnProperty.call(lDTO, 'mac') ||
      !Object.prototype.hasOwnProperty.call(lDTO, 'name')
    ) {
      logutil.warn('Bad DTO: ' + JSON.stringify(lDTO));
      reject();
      return;
    }
    models.Temperature.findOne({
      where: {
        mac: lDTO.mac
      }
    })
      .then(lTemperature => {
        if (lTemperature === null) {
          models.Temperature.create(lDTO)
            .then(() => {
              logutil.info(
                'Created temperature probe: ' + lDTO.name + ' with mac: ' + lDTO.mac
              );
              resolve();
            })
            .catch(err => {
              logutil.error('Error creating temperature probe:\n' + err);
              reject();
            });
        } else {
          logutil.warn('Probe already added: ' + lDTO.mac);
          reject();
        }
      })
      .catch(err => {
        logutil.error('Error saving temperatue:\n' + err);
        reject();
      });
  });
}

function handleNewReading(msg) {
  return new Promise(function(resolve, reject) {
    let lDTO = JSON.parse(msg.content.toString());
    if (
      !Object.prototype.hasOwnProperty.call(lDTO, 'mac') ||
      !Object.prototype.hasOwnProperty.call(lDTO, 'value')
    ) {
      logutil.warn('Bad DTO: ' + JSON.stringify(lDTO));
      reject();
      return;
    }
    models.Temperature.findOne({
      where: {
        mac: lDTO.mac
      }
    })
      .then(lTemperature => {
        if (lTemperature === null) {
          logutil.warn('Unknown temperature probe: ' + lDTO.mac);
          reject();
        } else {
          if (lTemperature.value !== lDTO.value) {
            lTemperature.update({ value: lDTO.value });
            mq.send('temperature.v1.valuechanged', lTemperature.toDTO());
          }
          resolve();
        }
      })
      .catch(err => {
        logutil.error('Error saving temperatue:\n' + err);
        reject();
      });
  });
}

function startMQ() {
  return new Promise(function(resolve, reject) {
    console.log('Connecting to MQ');
    mq
      .connect(process.env.MQ_ADDRESS, 'amq.topic')
      .then(() => {
        console.log('MQ Connected');
        return Promise.all([
          mq.recv('temperature', 'temperature.v1.createnew', handleCreateNew),
          mq.recv('temperature', 'temperature.v1.reading', handleNewReading)
        ]);
      })
      .then(() => {
        console.log('MQ Listening');
        resolve();
      })
      .catch(err => {
        console.warn(err);
        reject(err);
      });
  });
}

async function main() {
  console.log('Starting');
  await startMQ();
  await startDB();
  logutil.info('Temperature server started');

  Promise.all([
    addTempertureProbe({ mac: '28ff220b00150208', name: 'Cold Water' }),
    addTempertureProbe({ mac: '28ff6a02641403ed', name: 'Warm Water' }),
    addTempertureProbe({ mac: '28ff983d6414031a', name: 'Fermenter' })
  ])
    .then(() => {
      console.log('Test data created');
    })
    .catch(() => {
      console.log('Error during test data creation, could be normal if already created');
    });
}

main();
