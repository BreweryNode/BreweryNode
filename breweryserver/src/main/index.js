const Promise = require('bluebird');
var models;
let mq = require('brewerynode-common').mq;
const winston = require('winston');

function startDB() {
  return new Promise(function(resolve, reject) {
    models = require('./models');
    winston.silly('Syncing database');
    models.sequelize
      .sync({ force: true })
      .then(() => {
        winston.silly("Database sync'd");
        resolve();
      })
      .catch(err => {
        winston.warn(err);
        reject(err);
      });
  });
}

function startMQ() {
  return new Promise(function(resolve, reject) {
    winston.info('Connecting to MQ');
    mq
      .connect(process.env.MQ_ADDRESS, 'amq.topic')
      .then(() => {
        winston.info('MQ Connected');
        resolve();
      })
      .catch(err => {
        winston.warn(err);
        reject(err);
      });
  });
}

async function main() {
  winston.info('Starting');
  await startMQ();
  await startDB();
  winston.info('Starting handlers');
  require('./handlers');
  winston.info('Brewery server started');
}

const console = new winston.transports.Console({ level: 'silly' });
winston.add(console);
main();
