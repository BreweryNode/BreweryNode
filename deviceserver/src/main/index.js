const Promise = require('bluebird');
var models;
let mq = require('brewerynode-common').mq;
var logutil = require('brewerynode-common').logutil;

function startDB() {
  return new Promise(function(resolve, reject) {
    models = require('./models');
    logutil.silly('Syncing database');
    models.sequelize
      .sync({ force: true })
      .then(() => {
        logutil.silly("Database sync'd");
        resolve();
      })
      .catch(err => {
        logutil.warn(err);
        reject(err);
      });
  });
}

function startMQ() {
  return new Promise(function(resolve, reject) {
    console.log('Connecting to MQ');
    mq
      .connect(process.env.MQ_ADDRESS, 'amq.topic')
      .then(() => {
        console.log('MQ Connected');
        resolve();
      })
      .catch(err => {
        console.warn(err);
        reject(err);
      });
  });
}

async function main() {
  console.log('Starting');
  await startMQ();
  await startDB();
  logutil.info('Starting handlers');
  require('./handlers');
  logutil.info('Reading server started');
}

main();
