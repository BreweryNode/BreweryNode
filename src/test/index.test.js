describe('index', () => {
  it('Just logs main', () => {
    let output = '';
    let storeLog = inputs => {
      output += inputs;
    };
    const log = jest.fn(storeLog);
    console.log = log;
    require('../main/index');
    expect(output).toMatchSnapshot();
  });
});
