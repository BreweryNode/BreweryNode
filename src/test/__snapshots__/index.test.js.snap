// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index Just logs main 1`] = `"This is the main function."`;
